scientific/E3SM_Native_Grid_Test/E3SM_ne30_Native_Grid_Test.py

```diff
@@ -0,0 +1,160 @@
+# 
+# The grid file used is "ne30np4_latlon.091226.nc". To download this grid file, go to https://portal.nersc.gov/project/acme/hdavis/ and click on the "ne30np4_latlon.091226.nc" file whcih should automatically download via your browser. To facilitate running this notebook, the CMIP6 data file can also be downloaded from this same location.
+# 
+# Make sure both the data file and the grid file are in the same directory as this notebook before proceeding with running the notebook.
+
+# ## Open the CMIP6 data file
+# The variable "f" (for file) references the data file.
+
+# In[2]:
+
+
+f=cdms2.open("20180129.DECKv1b_piControl.ne30_oEC.edison.cam.h0.0500-01.nc")
+
+
+# ## Initialize the VCS canvas
+# In this example the variable "x" is the VCS canvas.
+
+# In[3]:
+
+
+x = vcs.init()
+
+
+# ## Set the DV3D settings
+# Choose various DV3D settings for the type of plot you want to make.
+
+# In[4]:
+
+
+dv3d = vcs.get3d_scalar()
+
+
+# In[5]:
+
+
+dv3d.ToggleVolumePlot = vcs.on
+dv3d.ToggleSphericalProj = vcs.on
+
+
+# In[6]:
+
+
+dv3d.Camera={'Position': (-161, -171, 279), 'ViewUp': (.29, 0.67, 0.68), 'FocalPoint': (146.7, 8.5, -28.6)}
+
+
+# ## Choose a variable to plot
+# The next line of code shows us all the data variables within the CMIP6 netCDF file.
+
+# In[7]:
+
+
+print(sorted(f.variables.keys()))
+
+
+# The variable we'll plot in this tutorial is temperature, T which is in degrees Kelvin. The next line assigns the temperature data (T) to the "v" variable.
+
+# In[8]:
+
+
+v = f["T"]
+
+
+# To ensure the color scale of our plot covers the minimum and maximum values of the data, we'll first determine min and max values, then modify the colormap scale.
+# 
+# Both VCS and Genutil have minmax functions. Here we'll use VCS's minmax function, but if you'd like to use Genutil's function instead, comment out the first line of code and un-comment the second and third lines. 
+
+# In[9]:
+
+
+vcs.minmax(v)
+
+#import genutil
+#genutil.minmax(v)
+
+
+# Modify the ScaleColormap to cover the min and max values of the data.
+
+# In[10]:
+
+
+dv3d.VerticalScaling = 1.5 
+dv3d.ScaleColormap = [184.0, 308.0, 1] 
+dv3d.ScaleTransferFunction =  [184.0, 308.0, 1]
+
+
+# ## Create the 3D plot
+
+# In[11]:
+
+
+x.plot( v, dv3d, grid_file="ne30np4_latlon.091226.nc")
+
+
+# If you want to interact with the 3D plot, save a python script from this notebook to run outside the notebook, then edit the python script to un-comment out the following line. Use a non-mesalib conda environment when running the Python script so you can interact with the 3D image in the window that opens after running the x.plot( v, dv3d ) command.
+
+# In[12]:
+
+
+x.interact()
+
+
+# ## Save a 2D image of the 3D plot
+# When working inside a notebook, saving the above image to a separte .png file could be useful if you want to look at the image outside of the notebook. The line of code below will save the plot to a .png file in the same directory as this notebook. 
```

Comment:
When working inside a notebook, saving the above image to a separate .png file could be useful if you want to look at the image outside of the notebook.

Reply:
Fixed.